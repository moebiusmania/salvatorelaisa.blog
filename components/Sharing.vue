<script setup lang="ts">
interface Props {
  url?: string;
}

const source = (url: string): string =>
  `https://github.com/moebiusmania/salvatorelaisa.blog/blob/main/content${url}.md`;

// const twitter = (url: string): string =>
//   `https://twitter.com/intent/tweet?text=${encodeURIComponent(
//     `https://salvatorelaisa.blog${url}`
//   )} @moebiusmania - `;

// const threads = (url: string): string =>
//   `https://threads.net/intent/post?text=${encodeURIComponent(
//     `https://salvatorelaisa.blog${url}`
//   )} @moebiusmania - `;

const bsky = (url: string): string =>
  `https://bsky.app/intent/compose?text=${encodeURIComponent(
    `https://salvatorelaisa.blog/post${url}`
  )} @moebiusmania.bsky.social - `;

const { url } = defineProps<Props>();
</script>

<template>
  <div>
    <a target="_blank" rel="nofollow noopener noreferrer" :href="bsky(url || '')">Parlane
      su Bluesky</a>
    <span>•</span>
    <!-- <a target="_blank" rel="nofollow noopener noreferrer" :href="twitter(url || '')">Parlane
      su X (<i>Twitter</i>)</a>
    <span>•</span> -->
    <a target="_blank" rel="nofollow noopener noreferrer" :href="source(url || '')">Codice
      sorgente su GitHub</a>
  </div>
</template>

<style scoped>
div {
  display: flex;
  flex-direction: column;
  padding: var(--sp-1) 0 var(--sp-3);
  align-items: center;
  justify-content: center;
  font-size: 0.875rem;
  line-height: 1.25rem;

  @media screen and (min-width: 768px) {
    flex-direction: row;
    align-items: start;
  }

  & a {
    text-decoration: underline;
    text-decoration-style: dotted;
    cursor: pointer;

    &:hover {
      color: var(--primary);
    }
  }

  & span {
    display: inline-block;
    padding: 0 var(--sp-1);
  }
}
</style>
