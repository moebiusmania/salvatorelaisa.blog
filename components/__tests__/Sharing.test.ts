import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import Sharing from "../Sharing.vue";

describe("Sharing", () => {
  // Snapshot test
  it("renders correctly", () => {
    const wrapper = mount(Sharing, {
      props: {
        url: "/blog/test-article",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  // Unit tests
  it("generates correct source URL", () => {
    const wrapper = mount(Sharing, {
      props: {
        url: "/blog/test-article",
      },
    });

    const sourceLink = wrapper.find('a[href*="github.com"]');
    expect(sourceLink.exists()).toBe(true);
    expect(sourceLink.attributes("href")).toBe(
      "https://github.com/moebiusmania/salvatorelaisa.blog/blob/main/content/blog/test-article.md"
    );
  });

  it("generates correct Threads URL", () => {
    const wrapper = mount(Sharing, {
      props: {
        url: "/blog/test-article",
      },
    });

    const threadsLink = wrapper.find('a[href*="threads.net"]');
    expect(threadsLink.exists()).toBe(true);
    expect(threadsLink.attributes("href")).toContain(
      "https://threads.net/intent/post?text="
    );
    expect(threadsLink.attributes("href")).toContain(
      encodeURIComponent("https://salvatorelaisa.blog/blog/test-article")
    );
  });

  it("handles undefined URL gracefully", () => {
    const wrapper = mount(Sharing);

    const sourceLink = wrapper.find('a[href*="github.com"]');
    const threadsLink = wrapper.find('a[href*="threads.net"]');

    expect(sourceLink.attributes("href")).toBe(
      "https://github.com/moebiusmania/salvatorelaisa.blog/blob/main/content.md"
    );
    expect(threadsLink.attributes("href")).toContain(
      encodeURIComponent("https://salvatorelaisa.blog")
    );
  });

  it("has correct security attributes on links", () => {
    const wrapper = mount(Sharing, {
      props: {
        url: "/blog/test-article",
      },
    });

    const links = wrapper.findAll("a");
    links.forEach((link) => {
      expect(link.attributes("target")).toBe("_blank");
      expect(link.attributes("rel")).toBe("nofollow noopener noreferrer");
    });
  });

  it("renders correct number of separator dots", () => {
    const wrapper = mount(Sharing, {
      props: {
        url: "/blog/test-article",
      },
    });

    const separators = wrapper.findAll("span");
    // Currently there should be 1 separator as Twitter is commented out
    expect(separators.length).toBe(1);
  });
});
