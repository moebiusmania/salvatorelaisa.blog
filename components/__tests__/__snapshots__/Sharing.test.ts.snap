// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sharing > renders correctly 1`] = `
"<div data-v-5d755431=""><a data-v-5d755431="" target="_blank" rel="nofollow noopener noreferrer" href="https://threads.net/intent/post?text=https%3A%2F%2Fsalvatorelaisa.blog%2Fblog%2Ftest-article @moebiusmania - ">Parlane su Threads</a><span data-v-5d755431="">•</span>
  <!-- <a target="_blank" rel="nofollow noopener noreferrer" :href="twitter(url || '')">Parlane
      su X (<i>Twitter</i>)</a>
    <span>•</span> --><a data-v-5d755431="" target="_blank" rel="nofollow noopener noreferrer" href="https://github.com/moebiusmania/salvatorelaisa.blog/blob/main/content/blog/test-article.md">Codice sorgente su GitHub</a>
</div>"
`;
